### ISSUE TITLE
cannot list secrets at the cluster scope

### ISSUE THREAD

**Issue by kiddo3** [Praqma/helmsman#83]
With version 1.6.0 I'm trying to deploy using a kubernetes user with permission to specific namespace "test", but a permission error occurs. In the dsf file I only configure the kubeContext.
Maybe I'm missing something or it's required the access at cluster scope?

```log
2018/09/19 13:56:06 INFO: Parsed YAML [[ deploy.yaml ]] successfully and found [ 2 ] apps.
2018/09/19 13:56:06 INFO: namespace validation -- a pre-installed Tiller is desired to be used in namespace [ test ].
2018/09/19 13:56:06 INFO: namespace validation -- Tiller is NOT desired to be deployed in namespace [ kube-system ].
2018/09/19 13:56:09 WARN: I could not create namespace [kube-system ]. It already exists. I am skipping this.
2018/09/19 13:56:10 WARN: I could not create namespace [test]. It already exists. I am skipping this.
2018/09/19 13:56:18 INFO: checking what I need to do for your charts ... 
2018/09/19 13:56:18 INFO: mapping the current helm state ...
2018/09/19 13:56:18 INFO: checking if any Helmsman managed releases are no longer tracked by your desired state ...
2018/09/19 13:56:19 No resources found.
Error from server (Forbidden): secrets is forbidden: User "system:serviceaccount:test:test-user" cannot list secrets at the cluster scope
```

**Comment by sami-alajrami** [comment]
@kiddo3 in the second line of the logs, it shows you are using your own (pre-deployed) Tiller in the test namespace .. so I am assuming you are using `useTiller`. This means that your pre-deployed Tiller is deployed without an appropriate service account. If you try `helm ls --tiller-namespace test` you should get the same forbidden error. 

Please check the [helm guide for deploying Tiller with service account](https://docs.helm.sh/using_helm/#role-based-access-control). Alternatively, you can let Helmsman deploy your Tiller and it will create the service account for you.

**Comment by kiddo3** [comment]
Hi @sami-alajrami,
I tried to run the command: ```helm ls --tiller-namespace test``` and is working fine.
Here is de yaml configuration:
```yaml
apps:
  nginx:
    chart: stable/nginx-ingress
    enabled: true
    name: nginx
    namespace: test
    tillerNamespace: test
    #valuesFile: helm-values/nginx_values.yaml
    version: 0.28.1
    wait: true
helmRepos:
  stable: https://kubernetes-charts.storage.googleapis.com
metadata:
  org: testing
namespaces:
  kube-system:
    installTiller: false
  test:
    protected: false
    useTiller: true
settings:
  kubeContext: kubernetes-dev-test
  storageBackend: secret
```
Thank you so much for your help.

**Comment by sami-alajrami** [comment]
@kiddo3 could you run helmsman with `--verbose` flag and share the logs (after verifying they don't contain any secrets) .. this will log the actual helm command triggering the error

**Comment by kiddo3** [comment]
here the log output:
```log
 _          _ 
| |        | | 
| |__   ___| |_ __ ___  ___ _ __ ___   __ _ _ __
| '_ \ / _ \ | '_ ` _ \/ __| '_ ` _ \ / _` | '_ \ 
| | | |  __/ | | | | | \__ \ | | | | | (_| | | | | 
|_| |_|\___|_|_| |_| |_|___/_| |_| |_|\__,_|_| |_|version: v1.6.0
A Helm-Charts-as-Code tool.


2018/09/20 08:24:02 VERBOSE: kubectl client version: v1.11.2
2018/09/20 08:24:02 VERBOSE: Helm client version: v2.10.0+g9ad53aa
2018/09/20 08:24:02 INFO: Parsed YAML [[ deploy-copy.yaml ]] successfully and found [ 1 ] apps.
2018/09/20 08:24:02 INFO: namespace validation -- Tiller is NOT desired to be deployed in namespace [ kube-system ].
2018/09/20 08:24:02 INFO: namespace validation -- a pre-installed Tiller is desired to be used in namespace [ test ].
2018/09/20 08:24:02 VERBOSE: kubectl config use-context kubernetes-dev-test
2018/09/20 08:24:02 VERBOSE: kubectl create namespace kube-system
2018/09/20 08:24:12 WARN: I could not create namespace [kube-system ]. It already exists. I am skipping this.
2018/09/20 08:24:12 VERBOSE: kubectl create namespace test
2018/09/20 08:24:13 WARN: I could not create namespace [test ]. It already exists. I am skipping this.
2018/09/20 08:24:13 VERBOSE: helm list --tiller-namespace test
2018/09/20 08:24:16 VERBOSE: helm repo add stable https://kubernetes-charts.storage.googleapis.com
2018/09/20 08:24:19 VERBOSE: helm repo update 
2018/09/20 08:24:21 VERBOSE: helm search stable/nginx-ingress --version 0.28.1 -l
2018/09/20 08:24:22 INFO: checking what I need to do for your charts ... 
2018/09/20 08:24:22 INFO: mapping the current helm state ...
2018/09/20 08:24:22 INFO: checking if any Helmsman managed releases are no longer tracked by your desired state ...
2018/09/20 08:24:22 VERBOSE: kubectl get secret --all-namespaces -l MANAGED-BY=HELMSMAN
2018/09/20 08:24:22 No resources found.
Error from server (Forbidden): secrets is forbidden: User "system:serviceaccount:test:test-user" cannot list secrets at the cluster scope
```

**Comment by sami-alajrami** [comment]
@kiddo3 it looks like your context (which is pre-setup, i.e. not created by helmsman) has the user test-user which has limited permissions. The failing command requires clusteradmin access (since helmsman is supposed to work on multiple namespaces )
How did you setup your context? is it a GKE cluster or other hosted-provider or on-prem?

In the meantime, using the `--keep-untracked-releases` option will skip this command. But you might run into the same error in other places.

**Comment by kiddo3** [comment]
It's an Azure AKS, it's a multitenant cluster so I won't be able to have access to the whole cluster only to deploy to some specific namespaces.

**Comment by sami-alajrami** [comment]
hmm .. as far as I can see, that check (checking untracked releases that should be cleaned) is the only command that will try to access the entire cluster. 
Could you please try with `--keep-untracked-releases` and see if you still into any other permission errors? .. this flag will stop helmsman from deleting releases if you remove them from your desired state.
We can change the `--all-namespaces` part of that kubectl command to cover only namespaces defined in the desired state file. But I want to verify if this would be the only permission problem to fix :)

**Comment by kiddo3** [comment]
I tried setting the parameter ```--keep-untracked-releases``` and now it's working :)

**Comment by sami-alajrami** [comment]
great .. the only other place where things can fail is if you ask helmsman to deploy Tiller to kube-system (which only makes sense if you have full cluster anyway!).

**Comment by kiddo3** [comment]
@sami-alajrami Thanks a lot for the clarification and for your help

**Comment by sami-alajrami** [comment]
fixed in v1.6.1 now will only check your defined namespaces which use or deploy Tiller.
